renderer.py

```diff
@@ -94,14 +100,21 @@ def init_shader(self):
       vert_out.smooth("VEC3", "cc_ck_scale")
       vert_out.smooth("FLOAT", "cc_k4")
       vert_out.smooth("FLOAT", "cc_k5")
+      vert_out.smooth("VEC2", "primDepth")
       vert_out.smooth("VEC4", "uv")
       vert_out.no_perspective("VEC2", "posScreen")
       vert_out.flat("VEC4", "tileSize")
       vert_out.flat("INT", "flags")
+      vert_out.flat("INT", "geoMode")
+      vert_out.flat("INT", "othermodeL")
```

Comment:
While unlikely that we are going to hit it, there is a limit in how many values you can pass to a fragment shader. It's also better to use vectors if possible since gl may treat a float as a vector internally.
So you can use e.g. IVEC2 for othermode

Reply:
good idea i didnt think about that, i will do that once i move these into the uniform buffer (should it be renamed if i pass in non cc data anyways?)

Reply:
thinking about it, if they are in the UBO anyways there is no need to pass them to the fragment shader manually since these settings don't differ between vertices/fragment.
You can directly access them like e.g. the tile settings do

Reply:
> thinking about it, if they are in the UBO anyways there is no need to pass them to the fragment shader manually since these settings don't differ between vertices/fragment. You can directly access them like e.g. the tile settings do

I just pushed a commit for that too, if you dont like it because of scope I can revert it